hudi-common/src/main/java/org/apache/hudi/common/model/RecordPayloadType.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import org.apache.hudi.common.config.EnumDescription;
+import org.apache.hudi.common.config.EnumFieldDescription;
+import org.apache.hudi.common.model.debezium.MySqlDebeziumAvroPayload;
+import org.apache.hudi.common.model.debezium.PostgresDebeziumAvroPayload;
+
+/**
+ * Payload to use for record.
+ */
+@EnumDescription("Payload to use for merging records")
+public enum RecordPayloadType {
+  @EnumFieldDescription("Provides support for seamlessly applying changes captured via Amazon Database Migration Service onto S3.")
+  AWS_DMS_AVRO(AWSDmsAvroPayload.class.getName()),
+
+  @EnumFieldDescription("Honors ordering field in both preCombine and combineAndGetUpdateValue.")
+  HOODIE_AVRO_DEFAULT(DefaultHoodieRecordPayload.class.getName()),
```

Comment:
Are these options expected to be used by users? Then there might be in-consistency for table config and write config, for write config, we still prefer the class name ?

Reply:
For write config, users can set both classname or enum. For compatibility reasons, I introduced a new config e.g. `WRITE_PAYLOAD_TYPE`, so that it does not break for users who are setting existing config. This PR ensures that the classnames don't get written to `hoodie.properties`. #8566 handles for write configs. It does not fix table properties. After this PR, #8566 should be rebased.